File: задача3_решение.md/TwitterUserService.py
import re

class TwitterUserService(UserService):
    def __init__(self):
        self.client = TwitterClient()
    
    def get_user_id(self, page_url: str) -> str:
        match = re.search(r"twitter.com/(\w+)", page_url)
        return match.group(1) if match else ""
    
    def get_user_name(self, user_id: str) -> str:
        return self.client.GetUserNameById(int(user_id))
    
    def get_friends(self, user_id: str) -> list['UserInfo']:
        subscribers = self.client.GetSubscribers(int(user_id))
        return [
            UserInfo(
                UserId=str(u.UserId),
                Name=self.client.GetUserNameById(u.UserId)
            ) for u in subscribers
        ]

File: задача3_решение.md/TwitterUserService.cs
public class TwitterUserService : UserService
{
    private readonly TwitterClient _client = new TwitterClient();

    protected override string GetUserId(string pageUrl)
    {
        var regex = new Regex("twitter.com/(.*)");
        return regex.Match(pageUrl).Groups[1].Value;
    }

    protected override string GetUserName(string userId)
    {
        return _client.GetUserNameById(long.Parse(userId));
    }

    protected override UserInfo[] GetFriends(string userId)
    {
        var subscribers = _client.GetSubscribers(long.Parse(userId));
        return subscribers.Select(u => new UserInfo 
        {
            UserId = u.UserId.ToString(),
            Name = _client.GetUserNameById(u.UserId)
        }).ToArray();
    }
}
